diff --git a/src/bluez-a2dp.h b/src/bluez-a2dp.h
index 4826a6f..715fd89 100644
--- a/src/bluez-a2dp.h
+++ b/src/bluez-a2dp.h
@@ -17,15 +17,15 @@
 
 #include "a2dp-codecs.h"
 
-const a2dp_sbc_t bluez_a2dp_sbc;
+extern const a2dp_sbc_t bluez_a2dp_sbc;
 #if ENABLE_MP3
-const a2dp_mpeg_t bluez_a2dp_mpeg;
+extern const a2dp_mpeg_t bluez_a2dp_mpeg;
 #endif
 #if ENABLE_AAC
-const a2dp_aac_t bluez_a2dp_aac;
+extern const a2dp_aac_t bluez_a2dp_aac;
 #endif
 #if ENABLE_APTX
-const a2dp_aptx_t bluez_a2dp_aptx;
+extern const a2dp_aptx_t bluez_a2dp_aptx;
 #endif
 
 #endif
diff --git a/src/bluez-iface.h b/src/bluez-iface.h
index 7d61638..a86a1bd 100644
--- a/src/bluez-iface.h
+++ b/src/bluez-iface.h
@@ -13,7 +13,7 @@
 
 #include <gio/gio.h>
 
-const GDBusInterfaceInfo bluez_iface_endpoint;
-const GDBusInterfaceInfo bluez_iface_profile;
+extern const GDBusInterfaceInfo bluez_iface_endpoint;
+extern const GDBusInterfaceInfo bluez_iface_profile;
 
 #endif
