From ad9e729231211c2ba437294a354966b86441988d Mon Sep 17 00:00:00 2001
From: Andrei Gherzan <andrei.gherzan@huawei.com>
Date: Fri, 10 Dec 2021 11:06:05 +0100
Subject: [PATCH] Dynamically load openssl.cnf for openssl-3.0

Upstream-status: Pending
Signed-off-by: Andrei Gherzan <andrei.gherzan@huawei.com>
Signed-off-by: Lans Zhang <jia.zhang@windriver.com>
---
 Make.rules | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/Make.rules b/Make.rules
index 38c7a22..69c57cb 100644
--- a/Make.rules
+++ b/Make.rules
@@ -117,13 +117,23 @@ getvar = $(shell if [ "$(1)" = "PK" -o "$(1)" = "KEK" ]; then echo $(1); else ec
 	$(CC) -S $(INCDIR) $(cflags) $(cppflags) -fno-toplevel-reorder -DBUILD_EFI -c $< -o $@
 
 %.crt:
-	$(OPENSSL) req -new -x509 -newkey rsa:2048 -subj "/CN=$*/" -keyout $*.key -out $@ -days 3650 -nodes -sha256
+	@if [ -s "$(OPENSSL_LIB)/ssl/openssl.cnf" ]; then \
+	    cfg="$(OPENSSL_LIB)/ssl/openssl.cnf"; \
+	else \
+	    cfg="$(OPENSSL_LIB)/ssl-3/openssl.cnf"; \
+	fi; \
+	$(OPENSSL) req -config $$cfg -new -x509 -newkey rsa:2048 -subj "/CN=$*/" -keyout $*.key -out $@ -days 3650 -nodes -sha256
 
 %.cer: %.crt
 	$(OPENSSL) x509 -in $< -out $@ -outform DER
 
 %-subkey.csr:
-	$(OPENSSL) req -new -newkey rsa:2048 -keyout $*-subkey.key -subj "/CN=Subkey $* of KEK/" -out $@ -nodes
+	@if [ -s "$(OPENSSL_LIB)/ssl/openssl.cnf" ]; then \
+	    cfg="$(OPENSSL_LIB)/ssl/openssl.cnf"; \
+	else \
+	    cfg="$(OPENSSL_LIB)/ssl-3/openssl.cnf"; \
+	fi; \
+	$(OPENSSL) req -config $$cfg -new -newkey rsa:2048 -keyout $*-subkey.key -subj "/CN=Subkey $* of KEK/" -out $@ -nodes
 
 %-subkey.crt: %-subkey.csr KEK.crt
 	$(OPENSSL) x509 -req -in $< -CA DB.crt -CAkey DB.key -set_serial 1 -out $@ -days 365
-- 
2.25.1

